### PR TITLE
Add option to enable active or passive mode

### PR DESCRIPTION
**Pull request by hannesdejager**
This adds functionality to libunftp to disable either Active Mode or Passive Mode or have both enabled. The default is to have only Passive mode (PASV) enabled since active mode is older and considered less secure.

If a client connects and uses PORT or PASV when it is disabled we return FTP code `502` (command not implemented)

### CODE REVIEW

File: src/server/ftpserver/options.rs [hunk 1]
```diff
@@ -240,3 +240,16 @@ impl Default for FailedLoginsPolicy {
         FailedLoginsPolicy::new(3, Duration::from_secs(120), FailedLoginsBlock::UserAndIP)
     }
 }
+
+/// The options for [Server.data_connection_mode](crate::Server::data_connection_mode).
+/// This allows to switch active / passive mode on or off.
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Default)]
+pub enum DataConnectionMode {
```
Comment (hannesdejager) [inline src/server/ftpserver/options.rs:247]: Alternative name is `ActivePassiveMode` which is probably more clear. What do you think @robklg ?

File: src/server/password.rs [hunk 2]
```diff
@@ -45,7 +45,7 @@ mod tests {
     const SECRET: &str = "supersecret";
 
     #[test]
-    fn password_obsures_display() {
+    fn password_obscures_display() {
```
Comment (hannesdejager) [inline src/server/password.rs:48]: unrelated fix.

File: src/server/ftpserver/options.rs (hunk 1 above)
Comment (robklg) [reply src/server/ftpserver/options.rs:247]: I agree `ActivePassiveMode` is clearer.

Comment (hannesdejager) [reply src/server/ftpserver/options.rs]: Renamed
